libraries/AP_Winch/AP_Winch_Daiwa.cpp

```diff
@@ -0,0 +1,227 @@
+#include <AP_Winch/AP_Winch_Daiwa.h>
+#include <GCS_MAVLink/GCS.h>
+
+extern const AP_HAL::HAL& hal;
+
+// true if winch is healthy
+bool AP_Winch_Daiwa::healthy() const
+{
+    // healthy if we have received data within 3 seconds
+    return (AP_HAL::millis() - data_update_ms < 3000);
+}
+
+void AP_Winch_Daiwa::init()
+{
+    // initialise rc input and output
+    init_input_and_output();
+
+    // initialise serial connection to winch
+    const AP_SerialManager &serial_manager = AP::serialmanager();
+    uart = serial_manager.find_serial(AP_SerialManager::SerialProtocol_Winch, 0);
+    if (uart != nullptr) {
+        // always use baudrate of 115200
+        uart->begin(115200);
+    }
+}
+
+void AP_Winch_Daiwa::update()
+{
+    // return immediately if no servo is assigned to control the winch
```

Comment:
I wonder if we should still attempt to read data from the winch.

Reply:
yes, I think it's best to read from the winch even if there is no output setup.  There's no harm in reading and it allows the user to see the telemetry output of the winch during the setup even if they've gotten the servox_function setup incorrectly.  I.e. they can fix each problem in any order.